### PR TITLE
Bah 1637

### PR DESCRIPTION
**Pull request by rohit-yawalkar**
This PR is about upgrading below versions
- openmrs-core to 2.4.2
- web-services to 2.29.0
- javax servlet to 4.0.1

Removed skipping test configuration
Resolved failing test cases

### CODE REVIEW

**Comment by gsluthra** [comment]
please provide summary/description of PR.

**Comment by rohit-yawalkar** [comment]
> please provide summary/description of PR.

Hi Gurpreet, 
Added the description, please review now

File: pom.xml [hunk 1]
```diff
@@ -42,11 +42,12 @@
 	</modules>
 
 	<properties>
-		<openMRSVersion>2.1.6</openMRSVersion>
-		<openMRSWebServicesVersion>2.24.0</openMRSWebServicesVersion>
+		<openMRSVersion>2.4.2</openMRSVersion>
+		<openMRSWebServicesVersion>2.29.0</openMRSWebServicesVersion>
```
Comment (angshu) [inline pom.xml:46]: this should be  bumped to. 2.33.0, or whatever we are using for https://github.com/Bahmni/openmrs-distro-bahmni/blob/master/distro/pom.xml#L46 (Notice there is no branch for 2.4.x or 2.5)

File: api/pom.xml [hunk 2]
```diff
@@ -68,6 +68,12 @@
 			<version>${log4jVersion}</version>
 			<scope>provided</scope>
 		</dependency>
+		<dependency>
+			<groupId>javax.servlet</groupId>
+			<artifactId>javax.servlet-api</artifactId>
+			<version>${javaxServletVersion}</version>
+			<scope>provided</scope>
```
Comment (angshu) [inline api/pom.xml]: if only required for tests, then scope should test

Comment (rohit-yawalkar) [reply api/pom.xml]: Angshu, 
scope of javax.servlet-api dependency changed to test

File: pom.xml (hunk 1 above)
Comment (rohit-yawalkar) [reply pom.xml:46]: Angshu,
we were going through 2.4.0 release notes and there they have mentioned web-services version 2.29.0, we are planning to 2.30.0 for 2.5.0 OMRS upgrade
https://wiki.openmrs.org/display/RES/Platform+Release+Notes+2.4.0

File: api/pom.xml (hunk 2 above)
Comment (rohit-yawalkar) [reply api/pom.xml]: Angshu,
Changed javax.servlet-api dependency scope to test

**Review by angshu — APPROVED** [review]

**Review by N0-man — APPROVED** [review]

**Pull request merged**